/all route and page created

evanhe1 opened the pull request.

ellieyhcheng commented on acm_url/templates/links.html:
```diff
@@ -0,0 +1,13 @@
+{% extends 'base.html' %} {% block content %}
+<h1>Most Popular URLs</h1>
+<ul class="container">
+  <div id="title">Vanity</div>
```
You should make this a class, since IDs are meant to be unique.

ellieyhcheng commented on requirements.txt:
```diff
@@ -17,4 +17,4 @@ six==1.15.0
 WTForms==2.3.3
-psycopg2
\ No newline at end of file
+psycopg2-binary
```
I don't think we should switch to the binary version of the library. At least if you think it's necessary/better, we should do this on a separate PR (and testing that nothing breaks with this switch). For now, move this change out of this PR.

evanhe1 replied:
that wasn't supposed to make it into the commit, sorry

ellieyhcheng reviewed (changes requested):
A couple of things:
1. We should clear the scrollbar on the cells. Check the overflow attributes of these elements.
![image](https://user-images.githubusercontent.com/15387908/149578211-f3ae9b08-652d-4416-839e-87f473e2fede.png)
2. It'd be better if the table is a fixed size that fills the screen (like in the mock up), rather than a variable with depending on the links.
3. The width of the vanity and visits columns should be wider so the right side isn't too short. But this should be fixed with 2). Nevertheless, put a right padding on the titles to center the text anyways. 
4. Maybe make the full URLs links that people can just click on to visit

evanhe1 commented:
-constrained table size through implementation of pagination
-hid scrollbars for urls
-added clickable links for urls
-constrained minimum element widths so full vanity remains always visible
-cleaned up css selectors

ellieyhcheng reviewed (changes requested):
- I think something went quite wrong with the CSS...
- ![image](https://user-images.githubusercontent.com/15387908/149599691-e43a404f-7e09-44a5-83ba-4e5c8385e4cd.png)
- Also, the table size isn't about pagination, it's about the physical width of the table. Right now it's too small, since it's fitting to the length of the URLs. It should be wider (reference the mock up).  
- The pagination is nice, though. Perhaps keep the links per page to at least 10? 
- Make the prev and next into "submit" buttons with forms (so that it would have the same styling as the other pages) you can reference `url.html` to see how that's done.

evanhe1 commented:
-pagination reimplemented with wtforms (I think there's a bit of unnecessary overhead compared to just using anchor elements)
-adjusted grid attributes to span entire width of parent container

ellieyhcheng reviewed (approved):
LGTM!

Re: WTForm, we can keep it as is to be consistent and since it works if you don't want to change it. You're right that it's less overhead and probably better accessibility wise. So if you don't mind making the changes, you can give the prev/next anchors a class and add that class to `style.css` with the submit button stylings block. Make sure it doesn't have the underlines though.

evanhe1 commented:
-prev and next buttons reimplemented with a elements with consistent styling
-button alignment adjusted so they stay in the same place even when one of them is not displayed
